### PR TITLE
Unify jormungandr API Types to allow better reusability

### PR DESCRIPTION
**Pull request by NicolasDP**
This provides some interfaces and standard representation of some types. As an example on how to use this, this PR unifies the Fragment's Logs API across Jormungandr (the node) and the integration tests.

`jormungandr-lib` will replace `jormungandr-utils` on the long run.

On the long run, this library will provide interfaces for all the objects displayed to users or used in public APIs.

this PR bootstrap the work for #434

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [workspace]
 members = [
   "jormungandr-utils",
+  "jormungandr-lib",
```
Comment (mzabaluev) [inline Cargo.toml]: Is this different in purpose from jormungandr-utils so that this crate proliferation is justified?

Comment (mzabaluev) [reply Cargo.toml]: Alternatively, -utils could be morphed into this.

Comment (NicolasDP) [reply Cargo.toml]: it is going to replace it.

Comment (NicolasDP) [reply Cargo.toml]: it was stated in the commit message of the first commit, but not repeated in the PR description:

> add introduce jormungandr-lib, a set of components to replace jormungandr-utils

Comment (mzabaluev) [reply Cargo.toml]: I'd just do it right away then.

Comment (NicolasDP) [reply Cargo.toml]: too much code change/churn at once

**Pull request merged**